enhancements/management-workload-partitioning.md

```diff
@@ -0,0 +1,549 @@
+be especially useful for resource-constrained enviornments, such as
+single-node production deployments, where the user wants to reserve
+most of the CPU resources for their own workloads and needs to
+configure OpenShift to run on a fixed number of CPUs within the host.
+
+One example of this use case is seen in telecommunication service
+providers implementation of a Radio Access Network (RAN). This use case
+is discussed in more detail below.
+
+## Motivation
+
+In constrained environments, management workloads, including the
+OpenShift control plane, need to be configured to use fewer resources
+than they might by default in normal clusters. After examining
+[various approaches for scaling the resource
+requests](https://github.com/openshift/enhancements/pull/628), we are
+reframing the problem to allow us to solve it a different way.
+
+Customers who want us to reduce the resource consumption of management
+workloads have a fixed budget of CPU cores in mind. We want to use
+normal scheduling capabilities of kubernetes to manage the number of
+pods that can be placed onto those cores, and we want to avoid mixing
+management and normal workloads there.
+
+### Goals
+
+* This enhancement describes an approach for configuring OpenShift
+  clusters to run with management workloads on a restricted set of
+  CPUs.
+* Clusters built in this way should pass the same Kubernetes and
+  OpenShift conformance and functional end-to-end tests as single-node
+  deployments that are not isolating the management workloads.
+* We have a goal of running on 4 hyperthreads today, but we do not
+  know what constraints we might be given in the future so we need a
+  solution that is not tied to the current known limit.
+* We want a general approach, that can be applied to all OpenShift
+  control plane components.
+
+### Non-Goals
+
+* This enhancement is focused on CPU resources. Other compressible
+  resource types may need to be managed in the future, and those are
+  likely to need different approaches.
+* This enhancement does not address non-compressible resource
+  requests, such as for memory.
+* This enhancement does not address ways to disable operators or
+  operands entirely.
+* Although the results of this enhancement may be useful for
+  single-node developer deployments, multi-node production
+  deployments, and Internet-of-things devices, those use cases are not
+  addressed directly in this document.
+* This enhancement does not address reducing actual utilization. There
+  is no expectation that a cluster configured to use a small number of
+  cores for management services would offer exactly the same
+  performance as the default. It must be stable and continue to
+  operate reliably, but may respond more slowly.
+* This enhancement assumes that the configuration of management
+  services is done as part of installing the cluster, and cannot be
+  changed later. Future enhancements may address enabling or
```

Comment:
This confuses me, but I'll comment on the details.

Reply:
@derekwaynecarr emphasized that doing it this way made the performance of the cluster predictable from the outside. I think some of the rest of us liked it because it reduced the complexity and scope. If we don't have to support turning this on day 2, we simplify the first version and can ignore some edge cases, for now.